src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java

```diff
@@ -28,32 +30,29 @@
 public final class DefaultServerTracingHandler implements ServerTracingHandler {
 
   private final Tracing tracing;
-  private final HttpServerHandler<ServerRequest, Response> handler;
+  private final HttpServerHandler<ServerRequest, ServerResponse> handler;
   private final TraceContext.Extractor<ServerRequest> extractor;
-  private final SpanNameProvider spanNameProvider;
   private final Logger logger = LoggerFactory.getLogger(DefaultServerTracingHandler.class);
   @Inject
-  public DefaultServerTracingHandler(final HttpTracing httpTracing, final SpanNameProvider spanNameProvider) {
+  public DefaultServerTracingHandler(final HttpTracing httpTracing) {
     this.tracing = httpTracing.tracing();
-    this.handler = HttpServerHandler.<ServerRequest, Response>create(httpTracing, new ServerHttpAdapter());
+    this.handler = HttpServerHandler.<ServerRequest, ServerResponse>create(httpTracing, new ServerHttpAdapter());
     this.extractor = tracing.propagation().extractor((ServerRequest r, String name) -> r.getHeaders().get(name));
-    this.spanNameProvider = spanNameProvider;
   }
 
   @Override
   public void handle(Context ctx) throws Exception {
     ServerRequest request = new ServerRequestImpl(ctx.getRequest());
     final Span span = handler.handleReceive(extractor, request);
-    span.name(spanNameProvider.spanName(request, Optional.empty()));
-    final Tracer.SpanInScope scope = tracing.tracer().withSpanInScope(span);
 
+    //create a scope so that we can @Inject a SpanCustomizer into
+    //handlers and add tags, annotations, etc.
+    final Tracer.SpanInScope scope = tracing.tracer().withSpanInScope(span);
```

Comment:
I don't actually know whether this is "a thing" - i.e. wanting to be able `@Inject` a `SpanCustomizer` into a handler and add tags, etc. I guess it might be useful if it's something that you don't want to implement as a general "policy" using something like a custom `HttpServerParser` ¯\\_(ツ)_/¯

Reply:
maybe harder hitting comment could be.. "place the scan in scope so that downstream code, such as logging can see the ids"